src/ChargeDrift/ChargeDrift.jl

```diff
@@ -118,6 +118,34 @@ function _add_fieldvector_diffusion!(step_vectors::Vector{CartesianVector{T}}, d
     end
     nothing 
 end
+function _add_fieldvector_diffusion_TiedWithMobility!(step_vectors::Vector{CartesianVector{T}}, done::Vector{Bool}, Δt::T, calculate_mobility::Function, current_pos::Vector{CartesianPoint{T}}, crystal_temperature::T, CC::Type{Electron})::Nothing where {T <: SSDFloat}
+    for n in eachindex(step_vectors)
+        if done[n] continue end
+
+        mu=calculate_mobility(current_pos[n], CC) # mobility_mu in m^2/V/s
+        D=mu*crystal_temperature*8.617e-5 # D = mobility_mu * k0*T_Deadlayer/e # D in m^2/s # k0=8.617e-5 eV/K

```

Comment:
Can you use the Boltzmann constant defined in `MaterialProperties.jl`?
https://github.com/JuliaPhysics/SolidStateDetectors.jl/blob/32851dcd3d458879dfcebd07c2352dc681dddf06/src/MaterialProperties/MaterialProperties.jl#L3-L7

You might need to adjust for the units by multiplying/dividing by the `elementary_charge`, which is also defined there:
```suggestion
        D=mu*kB*crystal_temperature/elementary_charge # D = mobility_mu * k0*T_Deadlayer/e # D in m^2/s
```

For reference:
```julia
using SolidStateDetectors
SolidStateDetectors.kB / SolidStateDetectors.elementary_charge
```
```
8.617333262145179e-5
```

Reply:
Sure!